### ISSUE TITLE
Reshape: what if |newShape| is a scalar number

### ISSUE THREAD

**Issue by zolkis** [webmachinelearning/webnn#452]
_Originally posted by @BruceDai in https://github.com/webmachinelearning/webnn/pull/446#discussion_r1294189344_
            
> Currently the spec only defines `newShape`  as **a sequence of {{nullable}} {{unsigned long}}** -- how about creating a separate PR where `|newShape| can also be a scalar [=number=]`, then Spec editors and more reviewers can easily review it and give feedbacks.

**Comment by inexorabletash** [comment]
Just to make sure I'm following along, this proposes:

```diff
    partial interface MLGraphBuilder {
-     MLOperand reshape(MLOperand input, sequence<unsigned long> newShape);
+     MLOperand reshape(MLOperand input, (unsigned long or sequence<unsigned long>) newShape);
    };
```

And the behavior of `builder.reshape(op, 123);` is equivalent to `builder.reshape(op, [123]);` ?

**Comment by huningxin** [comment]
> And the behavior of builder.reshape(op, 123); is equivalent to builder.reshape(op, [123]); ?

It seems confusing. I am not aware any examples to use scalar number to represent shape. @BruceDai , could you please clarify your proposal?

**Comment by wacky6** [comment]
I'm not aware of models that make use of scalar shape.

What is the semantic difference between `shape = 123` vs `shape = (123)` ?

Torch and Numpy seems to treat `reshape(scalar)` as if it's `reshape([scalar])`. The scalar invocation looks like a syntax sugar?

```Python
>>> np.random.rand(1,3,4,4).reshape(48).shape
(48,)   // <- The returned shape is a sequence of integers

>>> torch.rand((1,3,4,4)).reshape(48).shape
torch.Size([48])      // <- The returned shape is a sequence of integers
```

**Comment by zolkis** [comment]
@BruceDai PTAL: I wonder whether after #476 is this still an issue.

**Comment by BruceDai** [comment]
> @BruceDai , could you please clarify your proposal?

Actually this proposal was from @zolkis's #446.

> @BruceDai PTAL: I wonder whether after https://github.com/webmachinelearning/webnn/pull/476 is this still an issue.

In #476, it removed the step for |newShape| being a scalar, 

```spec
- If |newShape| is a scalar [=number=], set |outputShape| to the [=/list=] « 1 ».
``` 
so current the declarations of |newShape| match below algorithm steps.

**Comment by zolkis** [comment]
Thanks @BruceDai , then this can be closed.
